### ISSUE TITLE
Go router issues

### ISSUE THREAD

**Issue by thomtomdup** [Earlybyte/aad_oauth#278]
Hello.
Have you test it with go router without issue ?
Because in my case i have many issues.
Thank you.
Regards.

**Comment by mikedigit** [comment]
Hi,
If i can give you some pointer, I use this package in my app using go router without issue yet.
However, your message does not provide a lot of context around the encountered error, so I'm not sure where to go from there.

**Comment by thomtomdup** [comment]
Hello.

I use it for ADB2C and the response give issue for gorouter :
For  example in redirection from ADB2C to flutter app : 
`GoException:` no routes for location: state=7ipvjc4ZDjuJAgXTxXpfGbeyzm....`

**Comment by badr-o** [comment]
I'm writing to you here because I have a problem with go_router and I don't know how to configure the package with it. 
here's my error message: 
E/flutter (11711): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Exception: Could not push new route using provided navigatorKey, Because NavigatorState returned from provided navigatorKey is null. Please Make sure provided navigatorKey is passed to WidgetApp. This can also happen if at the time of this method call WidgetApp is not part of the flutter widget tree

my app_router : 
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import 'package:xxx/views/login_screen.dart';

final navigatorKey = GlobalKey<NavigatorState>();
class AppRouter {
  final GoRouter router = GoRouter(
    navigatorKey: navigatorKey,
    routes: [
      GoRoute(
        path: '/',
        builder: (context, state) => const LoginScreen(),
      ),
    ],
  );
}
my main : 
class MyApp extends StatelessWidget {
  final _appRouter = AppRouter();

  MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp.router(
      debugShowCheckedModeBanner: false,
      routerConfig: _appRouter.router,
      theme: AppTheme.theme,
    );
    // return MaterialApp(
    //   debugShowCheckedModeBanner: false,
    //   home: const LoginScreen(),
    //   theme: AppTheme.theme,
    //   navigatorKey: navigatorKey,
    // );
  }
}
to sum up, when I use flutter's basic navigation and put my login view and everything else in the main.dart file, everything works and I'm able to log in with my microsoft account, but I want to use go_router and I also want to separate my login view from main.dart.
knowing that in MaterialApp.router I can't add navigatorKey

thanks in advance

**Comment by ShmuelSterbach-Cammebys** [comment]
I am also having issues. The `Config` object requires a `GlobalKey<NavigatorState>`, and the only way to get that is with `router.configuration.navigatorKey` because `MaterialApp.router` doesn't have a `navigatorKey` param. Except when I use `router.configuration.navigatorKey`, the following error is thrown:
```log
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following assertion was thrown while finalizing the widget tree:
Multiple widgets used the same GlobalKey.
The key [LabeledGlobalKey<NavigatorState>#d2b44] was used by multiple widgets. The parents of those
widgets were:
- InheritedGoRouter(goRouter: Instance of 'GoRouter')
- SignInContent(state: SignInContentState#4b314)
A GlobalKey can only be specified on one widget at a time in the widget tree.

When the exception was thrown, this was the stack:
#0      BuildOwner._debugVerifyGlobalKeyReservation.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:flutter/src/widgets/framework.dart:3103:13)
#1      _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:633:13)
#2      BuildOwner._debugVerifyGlobalKeyReservation.<anonymous closure>.<anonymous closure> (package:flutter/src/widgets/framework.dart:3047:20)
#3      _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:633:13)
#4      BuildOwner._debugVerifyGlobalKeyReservation.<anonymous closure> (package:flutter/src/widgets/framework.dart:3042:36)
#5      BuildOwner._debugVerifyGlobalKeyReservation (package:flutter/src/widgets/framework.dart:3111:6)
#6      BuildOwner.finalizeTree.<anonymous closure> (package:flutter/src/widgets/framework.dart:3168:11)
#7      BuildOwner.finalizeTree (package:flutter/src/widgets/framework.dart:3250:8)
#8      WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:987:19)
#9      RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:457:5)
#10     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1325:15)
#11     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1255:9)
#12     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1113:5)
#13     _invoke (dart:ui/hooks.dart:312:13)
#14     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:383:5)
#15     _drawFrame (dart:ui/hooks.dart:283:31)
════════════════════════════════════════════════════════════════════════════════════════════════════

Another exception was thrown: Multiple widgets used the same GlobalKey.
Another exception was thrown: Multiple widgets used the same GlobalKey.

```

**Comment by ShmuelSterbach-Cammebys** [comment]
I was able to solve the issue by passing the NavigatorKey from my ShellRoute to the library:
```dart
  final _rootShellNavigatorKey = GlobalKey<NavigatorState>();
  final _rootNavigatorKey = GlobalKey<NavigatorState>();

  late final _router = GoRouter(
    navigatorKey: _rootNavigatorKey,
    routes: [
      ShellRoute(
        navigatorKey: _rootShellNavigatorKey,
        ...
      )
      ...
    ]
    ...
  );
  ...
  Config(
    navigatorKey: _rootShellNavigatorKey,
    clientId: ...,
  );
```

**Comment by mvukojicic** [comment]
I have the same problem but I am using Getx, "A GlobalKey was used multiple times inside one widget's child list", @ShmuelCammebys do you maybe have an idea how to solve it in Getx? I have a GetMaterial wrapper, and I am using getPages for routing.
